lib/shopify_app/session/in_memory_user_session_store.rb

```diff
@@ -1,14 +1,24 @@
 # frozen_string_literal: true
 module ShopifyApp
   class InMemoryUserSessionStore < InMemorySessionStore
-    def self.store(session, user)
-      id = super
-      repo[user.shopify_user_id] = session
-      id
-    end
+    class << self
+      def store(session, user)
+        id = super
+        repo[user.shopify_user_id] = session
+        id
+      end
+
+      def retrieve_access_scopes(id)
+        repo[id].extra[:scopes]
+      end
+
+      def retrieve_access_scopes_by_shopify_user_id(shopify_user_id)
```

Comment:
This interface seems weird to me. Why do we need to expand the interface like this? Don't we already return the scopes as part of the session?

Reply:
It forces the caller to know the internals of the `ShopifyAPI::Session` object. We don't have an `access_scopes` field on there so we have to pass it into the `extra` hash. So accessing it would be something like this:
```ruby
session.extra[:scopes]
```

Reply:
Then I think we should expand the `Session` interface to add an `access_scopes` attribute since it's a logical part of that object, imo.

Reply:
Covering that in this [PR](https://github.com/Shopify/shopify_api/pull/843)